src/main/app/jurisdiction/moreDetailsContent.ts

```diff
@@ -1,46 +1,84 @@
   <li class="govuk-list govuk-list--bullet">mamwlad eich mam os nad oedd eich rhieni wedi priodi, neu os oedd eich tad wedi marw cyn i chi gael eich geni</li></ul>
   Os ydych yn gadael eich mamwlad ac yn setlo mewn gwlad arall fel oedolyn, yna efallai bydd y wlad newydd yn dod yn ‘ddomisil o’ch dewis chi’.<br><br>
-  Os nad ydych chi’n siŵr am eich domisil, dylech gael cyngor cyfreithiol.`;
+  Os nad ydych chi’n siŵr am eich domisil, dylech gael cyngor cyfreithiol.`,
+};
 
-export const enResidual = (isDivorce: boolean): string => {
-  return `If you’re in a same-sex couple and if none of the other connections apply, the court may still have jurisdiction if:
-    <ul class="govuk-list govuk-list--bullet"><li class="govuk-list govuk-list--bullet">you ${
-      isDivorce ? 'married' : 'formed your civil partnership'
-    } in England or Wales and </li>
-    <li class="govuk-list govuk-list--bullet">it would be in the interests of justice for the court to consider the application. For example, your home country does not allow ${
-      isDivorce ? 'divorce' : 'civil partnerships to be ended'
-    } between same-sex couples</li></ul>`;
+export const cyResidual = (isDivorce: boolean, partner: string): Record<string, string> => {
+  return {
+    heading: 'Awdurdodaeth weddillol',
+    body: `Fel arfer, i fod yn gymwys ar gyfer awdurdodaeth weddillol, mae’n rhaid i’ch domisil chi neu ddomisil eich ${partner} fod yng Nghymru neu Loegr. Ni all y naill na’r llall ohonoch fod yn ddinesydd gwlad arall yn yr UE (ac eithrio Denmarc) na phreswylio’n arferol mewn gwlad arall yn yr UE (ac eithrio Denmarc).<br><br>
+   Hefyd, os ydych yn briod â rhywun o’r un rhyw, efallai eich bod yn gymwys ar gyfer awdurdodaeth weddillol os yw pob un o’r canlynol yn berthnasol:
+   <ul class="govuk-list govuk-list--bullet"><li class="govuk-list govuk-list--bullet">mi wnaethoch briodi eich gilydd yn y DU</li>
+   <li class="govuk-list govuk-list--bullet">nid yw’r naill na’r llall ohonoch yn ddinesydd gwlad arall yn yr UE (ac eithrio Denmarc) na’n preswylio’n arferol mewn gwlad arall yn yr UE (ac eithrio Denmarc)</li>
+   <li class="govuk-list govuk-list--bullet">byddai er lles cyfiawnder i’r llys ystyried y cais (gallai hyn fod yn berthnasol, er enghraifft, os nad yw’r wlad lle mae eich cartref yn caniatáu i gyplau o’r un rhyw ${
+     isDivorce ? 'gael ysgariad' : 'ddod â phartneriaeth sifil i ben'
+   })</li></ul>
+   Fodd bynnag, gall awdurdodaeth weddillol fod yn gymhleth. Os nad ydych chi’n siŵr a yw hyn yn berthnasol i chi, dylech gael cyngor cyfreithiol.`,
+  };
 };
 
 export const jurisdictionMoreDetailsContent = (
   connections: JurisdictionConnections[] | undefined,
+  language: Language,
   isDivorce: boolean,
+  partner: string,
   showAllConnectionTypes = false
 ): { text: { heading: string; body: string }[]; title: string } => {
+  const isEnglish: boolean = language === 'en';
```

Comment:
it might just be better to have the parameter as isEnglish and you can just pass true or false.